src/models/AccessControlUnit.ts

```diff
@@ -0,0 +1,318 @@
+     */
+    private uninternRule(rule: ListRule) {
+        /**
+         * Remove a rule from the map, there may be rules from different lists in the cache.
+         * We don't want to invalidate those.
+         * @param map A map of entities to rules.
+         */
+        const removeRuleFromMap = (map: Map<string, ListRule[]>) => {
+            const entry = map.get(rule.entity);
+            if (entry !== undefined) {
+                const newEntry = entry.filter(internedRule => internedRule.sourceEvent.event_id !== rule.sourceEvent.event_id);
+                if (newEntry.length === 0) {
+                    map.delete(rule.entity);
+                } else {
+                    map.set(rule.entity, newEntry);
+                }
+            }
+        };
+        if (rule.isGlob()) {
+            removeRuleFromMap(this.globRules);
+        } else {
+            removeRuleFromMap(this.literalRules);
+        }
+    }
+
+    /**
+     * Add a rule to the cache e.g. it was added to a policy list.
+     * @param rule The rule to add.
+     */
+    private internRule(rule: ListRule) {
+        /**
+         * Add a rule to the map, there might be duplicates of this rule in other lists.
+         * @param map A map of entities to rules.
+         */
+        const addRuleToMap = (map: Map<string, ListRule[]>) => {
+            const entry = map.get(rule.entity);
+            if (entry !== undefined) {
+                entry.push(rule);
+            } else {
+                map.set(rule.entity, [rule]);
+            }
+        }
+        if (rule.isGlob()) {
+            addRuleToMap(this.globRules);
+        } else {
+            addRuleToMap(this.literalRules);
+        }
+    }
+
+    /**
+     * Update the cache for a single `ListRuleChange`.
+     * @param change The change made to a rule that was present in the policy list.
+     */
+    private updataCacheForChange(change: ListRuleChange): void {
+        if (change.rule.kind !== this.entityType || change.rule.recommendation !== this.recommendation) {
+            return;
+        }
+        switch (change.changeType) {
+            case ChangeType.Added:
+            case ChangeType.Modified:
+                this.internRule(change.rule);
+                break;
+            case ChangeType.Removed:
+                this.uninternRule(change.rule);
+                break;
+            default:
+                throw new TypeError(`Uknown ListRule change type: ${change.changeType}`);
+        }
+    }
+
+    /**
+     * Update the cache for a change in a policy list.
+     * @param changes The changes that were made to list rules since the last update to this policy list.
+     */
+    private updateCache(changes: ListRuleChange[]) {
+        changes.forEach(this.updataCacheForChange, this);
+    }
+}
+
+export enum Access {
+    /// The entity was explicitly banned by a policy list.
+    Banned,
+    /// The entity did not match any allow rule.
+    NotAllowed,
```

Comment:
If you do not match any allow rule you will always be `NotAllowed`

Reply:
Unless you are saying that we have behaviour elsewhere that contradicts this where If there are no allow rules for a entity/recommendation combination, we create one that matches "*". But I don't think it does, since that rule from a user's perspective exists.